dateparser/search/search.py

```diff
@@ -1,212 +1,185 @@
+                reduced_text_candidate = None
+                if start_index >= 0:
+                    reduced_text_candidate = text[:start_index] + text[end_index:]
+            break
         else:
-            languages = [shortname]
-            to_parse = original
+            for splitter in _secondary_splitters:
+                secondary_split = re.split(
+                    "(?<! )[" + splitter + "]+(?! )", date_object_candidate
+                )
+                if secondary_split and len(secondary_split) > 1:
+                    reduced_text_candidate = " ".join(secondary_split)
+                    secondary_split_made = True
+
+            if not reduced_text_candidate:
+                is_previous_punctuation = False
+                for index, char in enumerate(date_object_candidate):
+                    if char in _punctuations:
+                        if is_previous_punctuation:
+                            double_punctuation_split = [
+                                text[: index - 1],
+                                text[index - 1:],
+                            ]
+                            reduced_text_candidate = " ".join(double_punctuation_split)
+                            break
+                        is_previous_punctuation = True
+                    else:
+                        is_previous_punctuation = False
+
+    if reduced_text_candidate:
+        reduced_text_candidate = reduced_text_candidate.strip(" .,:()[]-'")
+
+    if (deep_search or secondary_split_made) and not (
+        text == reduced_text_candidate and is_recursion_call
+    ):
+        if reduced_text_candidate and len(reduced_text_candidate) > 2:
+            returnable_objects = _joint_parse(
+                text=reduced_text_candidate,
+                parser=parser,
+                data_carry=returnable_objects,
+                is_recursion_call=True,
+            )
 
-        parser = DateDataParser(languages=languages, settings=settings)
-        parsed, substrings = self.parse_found_objects(parser=parser, to_parse=to_parse,
-                                                      original=original, translated=translated, settings=settings)
-        parser._settings = Settings()
-        return list(zip(substrings, [i[0]['date_obj'] for i in parsed]))
+    return returnable_objects
 
 
 class DateSearchWithDetection:
     """
-    Class which executes language detection of string in a natural language, translation of a given string,
-    search of substrings which represent date and/or time and parsing of these substrings.
+    Class which handles language detection, translation and subsequent generic parsing of
+    string representing date and/or time.
 
+    :return: A date search instance
     """
+
     def __init__(self):
-        self.loader = LocaleDataLoader()
-        self.available_language_map = self.loader.get_locale_map()
-        self.search = _ExactLanguageSearch(self.loader)
```

Comment:
The removal of `DateSearchWithDetection.search` is backward-incompatible.

Reply:
I think keeping the old objects around with their old names, logging a warning when used (i.e. by exposing them through a property whose getter logs a warning), would be ideal.